### PR TITLE
Build dotcms library with no deps

### PR DESCRIPTION
**Pull request by fmontes**
After Nx, this library was getting build with tons of extra dependencies.

## What I did here:

1. All the libraries and app (dotcms-ui) get the interface from local `dotcms-models`
2. The new path to `dotcms-models` is `@dotcms/dotcms-models`
3. I replaced `node-fetch` with `cross-fetch`, supports  Nodejs, browser and react native
4. `dotcms` lib now run the tests with `Jest`

image

### CODE REVIEW

File: angular.json [hunk 1]
```diff
@@ -496,6 +451,40 @@
...
             }
+        },
+        "dotcms": {
+            "projectType": "library",
+            "root": "libs/dotcms",
+            "sourceRoot": "libs/dotcms/src",
+            "prefix": "dotcms",
+            "architect": {
+                "build": {
+                    "builder": "@angular-devkit/build-ng-packagr:build",
```
Comment (fmontes) [inline angular.json:462]: Nx uses `@nrwl/angular:ng-packagr-lite` to build but it doesn't build (or I didn't find the option) the UMD version (that some node apps might need) so I had to change to angular's

File: libs/dotcms/jest.config.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+module.exports = {
```
Comment (fmontes) [inline libs/dotcms/jest.config.js:1]: I'm running the tests with Jest now. Same tests syntax doesn't change.

File: libs/dotcms-models/tsconfig.lib.json [hunk 3]
```diff
@@ -7,13 +7,16 @@
         "declarationMap": true,
         "inlineSources": true,
         "types": [],
-        "lib": ["dom", "es2018"]
+        "lib": ["dom", "es2018"],
+        "paths": {
+            "@dotcms/*": ["dist/libs/*"]
```
Comment (fmontes) [inline libs/dotcms-models/tsconfig.lib.json:12]: https://www.gitmemory.com/issue/nrwl/nx/602/496959942

**Review by ivorpad — DISMISSED** [review]

File: jest.config.js [hunk 4]
```diff
@@ -1,3 +1,3 @@
 module.exports = {
-    projects: []
+    projects: ['<rootDir>/libs/dotcms']
```
Comment (hmoreras) [inline jest.config.js:2]: what is this for ? why just one lib ?

File: libs/dotcms-models/README.md [hunk 5]
```diff
@@ -1,7 +1,3 @@
-# DotCMS Models
```
Comment (hmoreras) [inline libs/dotcms-models/README.md:1]: this is a TODO ?

Comment (fmontes) [reply libs/dotcms-models/README.md:1]: Ooops, needs to update this readme.

File: jest.config.js (hunk 4 above)
Comment (fmontes) [reply jest.config.js:2]: Is the only lib we have jest tests for. the app uses karma.

**Review by hmoreras — APPROVED** [review]

**Review by ivorpad — APPROVED** [review]

**Pull request merged**